theory-test-admission-candidate-app/app/server/services/candidateCheckBookingService.js

```diff
@@ -1,21 +1,61 @@
 const AWS = require('aws-sdk');
+const moment = require('moment');
+const logger = require('logger');
+
+const bookingStepFunction = new AWS.StepFunctions();
 
 export default class CandidateCheckBookingService {
 
-	checkCandidateBooking() {
-
-		return true;
-		// const bookingStepFunction = new AWS.StepFunctions();
-		// const params = {
-		// 	stateMachineArn: process.env.SFN_START_CANDIDATE_ADMISSION_ARN,
-		// 	input: { DrivingLicenceNumber: this.candidateDLN, AdmissionId: this.admissionID, ReceivedDate: new Date() }
-		// };
-		// bookingStepFunction.startExecution(params, (err, data) => {
-		// 	if (err) {
-		// 		return false; // an error occurred
-		// 	}
-		// 	return data.HasBooking; // successful response
-		// });
+	static checkCandidateBooking(admissionId, candidateDLN, callback) {
+
+		logger.info('Check candidate booking executed');
+		const currentDate = moment(Number.parseInt(new Date().getTime(), 10)).toISOString(true);
+
+		const dln = `"DrivingLicenceNumber": "${candidateDLN}"`;
+		const aId = `"AdmissionId": "${admissionId}"`;
+		const date = `"Date": "${currentDate}"`;
+		const completeInput = `{ "Request": {${dln}, ${aId}, ${date}} }`;
+
+		const params = {
+			stateMachineArn: process.env.SFN_START_CANDIDATE_ADMISSION_ARN,
+			input: completeInput
+		};
+		this.doStartExecution(params, (response) => {
+			logger.info('Start execution executed and returned: ', response);
+			const descParams = {
+				executionArn: response.executionArn,
+			};
 
+			new Promise((r, j) => {
```

Comment:
Never be tempted to abbreviate parameter names like this.
Tomorrow it will be unreadable.